presidio-image-redactor/tests/test_image_analyzer_engine.py

```diff
@@ -113,3 +113,38 @@ def test_given_dif_len_entities_then_map_analyzer_returns_correct_outputand_len(
 
     assert len(expected_result) == len(mapped_entities)
     assert expected_result == mapped_entities
+
+
+@pytest.mark.parametrize(
+    "ocr_threshold, expected_length",
+    [(-1, 9), (50, 7), (80, 2), (100, 0)],
+)
+def test_threshold_ocr_result_returns_expected_results(
+    image_analyzer_engine, ocr_threshold, expected_length
+):
+    # Assign
+    ocr_result = {}
+    ocr_result["text"] = [
+        "",
+        "Homey",
+        "Interiors",
+        "was",
+        "created",
+        "by",
+        "Katie",
+        "",
+        "Cromley.",
+    ]
+    ocr_result["left"] = [143, 143, 322, 530, 634, 827, 896, 141, 141]
+    ocr_result["top"] = [64, 67, 67, 76, 64, 64, 64, 134, 134]
+    ocr_result["width"] = [936, 160, 191, 87, 172, 51, 183, 801, 190]
+    ocr_result["height"] = [50, 47, 37, 28, 40, 50, 40, 50, 50]
+    ocr_result["conf"] = [-1, 99.5, 92.3, 42.7, 66.1, 51.2, 79.7, 64.0, 70.3]
+
+    # Act
+    test_filtered = image_analyzer_engine.threshold_ocr_result(
```

Comment:
Consider adding an integration test as well. As discussed, I'm not sure this code would not work (haven't tested it though) because the `ocr_threshold` parameter would throw an error in `AnalyzerEngine.analyze`).

Reply:
It does work given the removal of `ocr_threshold` from `**kwargs` as defined in `ImageAnalyzerEngine.analyze` (the integration tests run successfully and I have tested separately as well), but let's see the outcome of the earlier comment before adjusting this.

Reply:
Great, I missed that part. Thanks!